### ISSUE TITLE
support react-native/expo discussion

### ISSUE THREAD

**Issue by peterpme** [onsol-labs/tld-parser#5]
Hi there, cross-posting this from https://github.com/coral-xyz/backpack/pull/3379

i temporarily removed tldparser (https://github.com/coral-xyz/backpack/pull/3365) bc it doesn't work on mobile via expo

we can move this discussion over to your repo. in the mean time, here's what we need:

1) [getHashedName](https://github.com/onsol-labs/tld-parser/blob/main/src/utils.ts#L67) should be an async function.

this change should have no impact on consumers of the library. [in the 2 places that its called](https://github.com/search?q=repo%3Aonsol-labs%2Ftld-parser%20getHashedName&type=code), there are already async functions included

wrapping it in node's utils.promisify should work great and have no impact on the overhead

2) include a [separate index.native.js ](https://github.com/LinusU/react-native-get-random-values) (a diff example linked so you can see how) file that can polyfill `crypto.createHash` using `expo-crypto`. expo-crypto will work with both react-native and crypto

a separate index.native.js file will be consumed by expo/react-native automatically without having an impact on your web users

the following code should be functionally equivalent:

```ts
const encryptedActionSecret1 = crypto
      .createHash("sha256")
      .update(bytes, "utf-8")
      .digest("utf-8") as bytes;

    const encryptedActionSecret2 = (await Crypto.digestStringAsync(
      Crypto.CryptoDigestAlgorithm.SHA256,
      Buffer.from(bytes, "utf-8").toString()
    )) as bytes;
```

if you have any questions please lmk. thank you!

**Comment by crypt0miester** [comment]
hello peter, I added the async in point 1. that was clear from my end. 

but the second part is over my head.

**Comment by tonivlad** [comment]
fixed in tldparser version 0.5.1

replied more here: https://github.com/coral-xyz/backpack/pull/3501#issue-1647778902
